### PR TITLE
add nodes to project state doc

### LINKED ISSUE #5648: Add node/resource types to project state doc
### Contributions

- [x] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

From @dbeatty10 :

The listing of node / resource types [here](https://docs.getdbt.com/reference/global-configs/resource-type) has hyperlinks, so it seems useful to add them to [this](https://docs.getdbt.com/docs/dbt-cloud-apis/project-state#affected-states-by-node-type) table as well.

We could probably add the ones from [here](https://docs.getdbt.com/reference/global-configs/resource-type) that are missing in this table (but I didn't in this PR, so we'd need to create a separate issue and/or PR):
- [Analysis](https://docs.getdbt.com/docs/build/analyses)
- [Saved query](https://docs.getdbt.com/docs/build/saved-queries) (which includes [exports](https://docs.getdbt.com/docs/build/saved-queries#configure-exports))
- [Unit test](https://docs.getdbt.com/docs/build/unit-tests)

### What part(s) of the page would you like to see updated?

https://docs.getdbt.com/docs/dbt-cloud-apis/project-state#affected-states-by-node-type

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr continue's @dbeatty10 's [pr](https://github.com/dbt-labs/docs.getdbt.com/pull/5641) and adds Analysis Saved query, and Unit tests to the nodes table.

Resolves #5648

### CODE REVIEW

File: website/docs/docs/dbt-cloud-apis/project-state.md [hunk 1]
```diff
@@ -61,16 +61,20 @@ Most Discovery API use cases will favor the _applied state_ since it pertains to
 
 | Node                                          | Executed in DAG  | Created by execution | Exists in database | Lineage               | States               |
 |-----------------------------------------------|------------------|----------------------|--------------------|-----------------------|----------------------|
-| [Model](/docs/build/models)                   | Yes              | Yes                  | Yes                | Upstream & downstream | Applied & definition |
-| [Source](/docs/build/sources)                 | Yes              | No                   | Yes                | Downstream            | Applied & definition |
-| [Seed](/docs/build/seeds)                     | Yes              | Yes                  | Yes                | Downstream            | Applied & definition |
-| [Snapshot](/docs/build/snapshots)             | Yes              | Yes                  | Yes                | Upstream & downstream | Applied & definition |
+| [Analysis](/docs/build/analyses)   	        | No               | No                   | No                 | N/A 	               | Definition 	      |
```
Comment (dbeatty10) [inline website/docs/docs/dbt-cloud-apis/project-state.md]: Based on the diagram [here](https://github.com/dbt-labs/docs.getdbt.com/issues/4492#issuecomment-1881658603), _Analysis_ looks to be the same as _Exposure_, which is "Upstream":

```suggestion
| [Analysis](/docs/build/analyses)   	        | No               | No                   | No                 | Upstream            | Definition 	      |
```

File: website/docs/docs/dbt-cloud-apis/project-state.md [hunk 2]
```diff
@@ -61,16 +61,20 @@ Most Discovery API use cases will favor the _applied state_ since it pertains to
...
 | [Group](/docs/build/groups)                   | No               | No                   | No                 | Downstream            | Definition           |
 | [Macro](/docs/build/jinja-macros)             | Yes              | No                   | No                 | N/A                   | Definition           |
+| [Metric](/docs/build/metrics-overview)     	| No               | No                   | No                 | Upstream & downstream | Definition           |
+| [Model](/docs/build/models)                   | Yes              | Yes                  | Yes                | Upstream & downstream | Applied & definition |
+| [Saved query](/docs/build/saved-queries)      | Yes              | Yes                  | Yes                | Upstream & downstream | Definition 	      |
+| [Seed](/docs/build/seeds)                     | Yes              | Yes                  | Yes                | Downstream            | Applied & definition |
+| [Semantic model](/docs/build/semantic-models) | No               | No                   | No                 | Upstream & downstream | Definition           |
```
Comment (dbeatty10) [inline website/docs/docs/dbt-cloud-apis/project-state.md:75]: Does the _Saved query_ lineage include "Downstream"?

Comment (mirnawong1) [reply website/docs/docs/dbt-cloud-apis/project-state.md:75]: confirmed by eng that Saved Queries" are not currently exposed through the Discovery API and not exposed in upstream or downstream lineage yet

**Comment by mirnawong1** [comment]
hey @dbeatty10 , thanks so much for your review! I've tweaked this to remove saved queries as eng confirmed it's not available in the disco api. whenever you have a chance, can you re-review? thankyou!

File: website/docs/docs/dbt-cloud-apis/project-state.md (hunk 2 above)
Comment (nataliefiann) [reply website/docs/docs/dbt-cloud-apis/project-state.md:75]: Hiya @mirnawong1 

I'm probably reading this wrong I wanted to ask in the **Affected states by node type** table for Saved Queries - under the Lineage header it says **Upstream & downstream** with the info you found out from eng, should that read **N/A**, **not exposed in upstream or downstream lineage** etc.

Kind Regards

**Review by nataliefiann — APPROVED** [review]
Hiya @mirnawong1  
I've had a look at the PR for you and it's great! The links work perfectly. I left a small note (but I think I may be reading it wrong, too deeply...)

Kind Regards

**Pull request merged**